rsc/src/main/scala/rsc/outline/Synthesizer.scala

```diff
@@ -175,8 +184,19 @@ final class Synthesizer private (
     }
     val ret = {
       val core = tree.id
-      if (tparams.isEmpty) Some(core)
-      else Some(TptParameterize(core, tparams.map(_.id.asInstanceOf[TptId])))
+      // Convert the implicit class's type parameters to the factory method return type parameters.
+      // This is necessary because the type parameters of a class or method definition can include
+      // constraints like upper and lower bounds, while a method return type can only include
+      // concrete type names and type variables that were bound in the method's type parameters.
+      val retTparams = tparams.map(_.id match {
+        case tptId: TptId => tptId
+        // AnonId represents `_`, which doesn't make sense as a type parameter in the return type;
+        // replace it by `Any`, i.e., `_root_.scala.Any`
+        case anonId: AnonId => TptId("Any").withSym(AnyClass)
+        case nonTptNamedId: NamedId => crash(nonTptNamedId.pos, nonTptNamedId)
```

Comment:
I don't think this is exhaustive; we can maybe say `case otherId =>`

Reply:
CI was failing because this was a non-exhaustive match without the `NamedId` fallback, but it is passing now. Here's why:
1. `TypeParam#id` has type `UnambigId`
1. `UnambigId` is a sealed trait with two implementations, `AnonId` and `NamedId`
1. `NamedId` is a sealed trait with four implementations: `CtorId`, `PatId`, `TermId`, `TptId`